### PR TITLE
Tighten exception handling in build_id_messages

### PR DESCRIPTION
**Pull request by jasonleinbach-wf**
#### Changes:
- Remove function-wide try/catch
- Add more specific try/catch/continues to lines that can throw exceptions
- add readme to non_library_plugins so git checks it out.
- update gitignore.

Unit tests not applicable here - functionally equivalent to before, and not very testable.

This is based on @brianmyers-wf 's 2883 branch.  

Before:
Arelle messages catalog 2.59 secs, 162 formula files, 1287 messages

After:
Arelle messages catalog 1.97 secs, 162 formula files, 1287 messages

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

File: non_library_plugins/README.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+This directory is included and purposefully left empty.  Had to add a file in it so that git checks it out.å
```
Comment (derekgengenbacher-wf) [inline non_library_plugins/README.txt]: Nit: Do we need that funky character at the end of the line?

**Comment by derekgengenbacher-wf** [comment]
+1

File: non_library_plugins/README.txt (hunk 1 above)
Comment (jasonleinbach-wf) [inline non_library_plugins/README.txt]: We do not - typo.  Good catch.

**Comment by derekgengenbacher-wf** [comment]
+1

**Comment by brianmyers-wf** [comment]
+1

**Comment by chrislococo-wf** [comment]
#### QA Steps:
- Manual testing: Generation of messages succeeded.
- Flake8: 

```
Not yet. TODO.
```
- Nosetest result: 

```
(builder)  chrislococo [arelle_builder] (master) $ nosetests test/
...........
----------------------------------------------------------------------
Ran 11 tests in 0.010s

OK
```

#### Result: QA +10

**Pull request merged**